lib/new_relic/agent/instrumentation/async_http.rb

```diff
@@ -0,0 +1,31 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'async_http/instrumentation'
+require_relative 'async_http/chain'
+require_relative 'async_http/prepend'
+
+DependencyDetection.defer do
+  named :'async_http'
+
+  depends_on do
+    # The class that needs to be defined to prepend/chain onto. This can be used
+    # to determine whether the library is installed.
+    defined?(Async::HTTP)
+    # Add any additional requirements to verify whether this instrumentation
+    # should be installed
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing async_http instrumentation')
+
+    require 'async/http/internet'
+    require 'new_relic/agent/http_clients/async_http_wrappers'
```

Comment:
The first `require` is needed here for the code block that follows it. As for the wrappers file, I recommend moving that one to `async_http/instrumentation.rb` to make it more clear that it's that code that needs it.

Reply:
Moved in https://github.com/newrelic/newrelic-ruby-agent/pull/2272/commits/fef6ba793936f2d306009c643c5ebdc316704704